src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/TransactionExpirationQueueConsumerTests.kt

```diff
@@ -3480,11 +3480,74 @@ class TransactionExpirationQueueConsumerTests {
       verify(refundRetryService, times(0)).enqueueRetryEvent(any(), any(), any())
     }
 
+    @Test
+    fun `messageReceiver should not perform REFUND when NPG has already refund the transaction`() {
+      val events =
+        listOf(
+          transactionActivateEvent(npgTransactionGatewayActivationData()),
+          transactionAuthorizationRequestedEvent(
+            TransactionAuthorizationRequestData.PaymentGateway.NPG,
+            npgTransactionGatewayAuthorizationRequestedData()))
+      setupTransactionStorageMock(events)
+      given(checkpointer.success()).willReturn(Mono.empty())
+      given { authorizationStateRetrieverService.getOrder(any()) }
+        .willAnswer {
+          OrderResponseDto()
+            .orderStatus(OrderStatusDto().lastOperationType(OperationTypeDto.REFUND))
+            .addOperationsItem(
+              OperationDto()
+                .orderId(UUID.randomUUID().toString())
+                .operationType(OperationTypeDto.AUTHORIZATION)
+                .operationResult(OperationResultDto.AUTHORIZED)
+                .paymentEndToEndId(UUID.randomUUID().toString())
+                .operationTime(ZonedDateTime.now().toString()))
+            .addOperationsItem(
+              OperationDto()
+                .orderId(UUID.randomUUID().toString())
+                .operationType(OperationTypeDto.REFUND)
+                .operationResult(OperationResultDto.VOIDED)
+                .paymentEndToEndId(UUID.randomUUID().toString())
+                .operationTime(ZonedDateTime.now().toString()))
+            .toMono()
```

Comment:
duplicate this test with operation items switched. seeing reduce implementation we expect both NPG responses to be considered as a refunded transaction so better to cover it in tests too

Reply:
Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/269/commits/bd7dc36b21438a31a82c524f0963afd442010638